src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -214,6 +214,33 @@ class ReferenceAdapter {
 	}
 }
 
+class ExternalReferenceAdapter {
+
+	constructor(private _provider: vscode.ExternalReferenceProvider) {
+	}
+
+	provideExternalReferences(query: modes.ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: modes.IReferenceInformation[]) => void): TPromise<modes.IReferenceInformation[]> {
+		const convertReferences = (references: vscode.ReferenceInformation[]) => {
+			if (Array.isArray(references)) {
+				return references.map(reference => {
+					return <modes.IReferenceInformation>{
```

Comment:
> you can write this shorter without a return

Not sure what you mean? If I just delete the return I get a compile error.

> don't think the cast is needed

It isn't needed, but I think it is nice for readability since it is just a plain object.